src/components/Auth/ResetPassword.tsx

```diff
@@ -0,0 +1,157 @@
+import React from 'react';
+import { useState, useCallback, useEffect } from 'react';
+import { useHistory } from 'react-router-dom';
+import qs from 'query-string';
+
+import { Form, Typography, Input, Button, notification } from 'antd';
+import { useAuth } from '../../hooks';
+import styled from 'styled-components';
+
+function ResetPassword() {
+  const { authenticating, onPasswordReset, isPasswordResetted } = useAuth();
+  const [password, setPassword] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState(' ');
+  const [enableSubmitButton, setEnableSubmitButton] = useState(false);
+
+  const history = useHistory();
+
+  const q = qs.parse(document.location.search);
+  const token = q.token as string;
+  const login = q.email as string;
+
+  const handlePasswordChange = useCallback((e: any) => {
+    setPassword(e.target.value);
+  }, []);
+  const handleconfirmPasswordChange = useCallback((e: any) => {
+    setConfirmPassword(e.target.value);
+  }, []);
+
+  const resetSuccessfulNotification = () => {
+    notification['success']({
+      message: 'Password has been successfully changed.',
+      description: 'Your password has been successfully changed. You can log in with your new password now.',
+      duration: 3.5,
+      // close model too
+    });
+  };
+
+  const handlePasswordReset = useCallback(() => {
+    if (password === confirmPassword) {
+      onPasswordReset({ login, password, token });
+    }
+  }, [login, token, password, confirmPassword, onPasswordReset]);
+
+  async function validatePassword(password: string): Promise<void> {
+    setEnableSubmitButton(false);
+    if (!/(?=.*[A-Z])(?=.*[a-z]).*/.test(password)) {
```

Comment:
Why are you copy pasting logic AGAIN? This will NOT be tolerated. Extract into common function